internal/channel/channel.go

```diff
@@ -13,38 +19,110 @@ type Channel struct {
 	Type   string `db:"type"`
 	Config string `db:"config" json:"-"` // excluded from JSON config dump as this may contain sensitive information
 
-	plugin Plugin
+	Logger *zap.SugaredLogger
+	cmd    *exec.Cmd
+	rpc    *rpc.RPC
+	mu     sync.Mutex
 }
 
-func (c *Channel) GetPlugin() (Plugin, error) {
-	if c.plugin == nil {
-		f := Channels[c.Type]
-		if f == nil {
-			return nil, fmt.Errorf("unknown channel type %q", c.Type)
-		}
+func (c *Channel) StartPlugin(pluginDir string) error {
+	c.mu.Lock()
+	defer c.mu.Unlock()
 
-		p, err := f(c.Config)
-		if err != nil {
-			return nil, fmt.Errorf("error initializing channel type %q: %w", c.Type, err)
-		}
+	if c.cmd != nil {
+		return nil
+	}
+
+	if matched, _ := regexp.MatchString("^[a-zA-Z0-9]*$", c.Type); !matched {
+		return fmt.Errorf("channel type must only contain a-zA-Z0-9, %s given", c.Type)
+	}
```

Comment:
Prefer `%q` when arbitrary strings (as in you don't exactly know what can be in the string) as this automatically escapes most strange things.